File: Divide Players Into Teams of Equal Skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
    sort(skill.begin(),skill.end());
    int i=0;
    int j=skill.size()-1;
    int ans=skill[i] + skill[j];
    long long chemistry= skill[i] * skill[j];
    i=1;
    j-=1;
    while(i<j){
     
        if (skill[i] + skill[j]!=ans)
          return -1;
        else
          chemistry=chemistry+(skill[i] * skill[j]); 
    i++;
    j--;
    }
        
    return chemistry;
   }
    
};

File: progress sheet/divide-players-into-teams-of-equal-skill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        chemistry = 0
        sum = 0
        i = 0
        j = len(skill) - 1
        skill.sort()
        while i <= j:
            if i == 0:
                sum = skill[i] + skill[j]
            elif sum != skill[i] + skill[j]:
                chemistry = -1
                break
            chemistry += (skill[i] * skill[j])
            i += 1
            j -= 1
        return chemistry

